heat/core/statistics.py

```diff
@@ -1100,6 +1132,231 @@ def mpi_argmin(a, b, _):
 MPI_ARGMIN = MPI.Op.Create(mpi_argmin, commute=True)
 
 
+def percentile(x, q, axis=None, interpolation="linear", keepdim=False):
+    """
+    Compute the q-th percentile of the data along the specified axis.
+    Returns the q-th percentile(s) of the tensor elements.
+
+    Parameters:
+    -----------
+
+    a : ht.tensor
+    Input tensor
+
+    q : ht.tensor, torch.tensor, scalar, or list of scalars
+    Percentile or sequence of percentiles to compute. Must belong to the interval [0, 100].
+
+    axis : int, or None, optional #TODO tuple of ints
+    Axis along which the percentiles are computed. Default is None.
+
+    #TODO: out, optional. Output buffer.
+
+    interpolation : {‘linear’, ‘lower’, ‘higher’, ‘midpoint’, ‘nearest’}, optional
+    Interpolation method to use when the desired percentile lies between two data points i < j:
+
+        ‘linear’: i + (j - i) * fraction, where fraction is the fractional part of the index surrounded by i and j.
+        ‘lower’: i.
+        ‘higher’: j.
+        ‘nearest’: i or j, whichever is nearest.
+        ‘midpoint’: (i + j) / 2.
+
+    keepdim : bool, optional
+    If True, the axes which are reduced are left in the result as dimensions with size one.
+    With this option, the result can broadcast correctly against the original array a.
+
+    Returns:
+    --------
+
+    percentile : process-local ht.tensor
+
+    """
+
+    def local_percentile(data, axis, indices):
+        """
+        Process-local percentile calculations
+
+        Input:
+        ------
+        data : ht.tensor
```

Comment:
I need the halo... but you're right, it should be torch-in, torch-out. I'll fix this

Reply:
Well actually there is a lot of torch functionality that could be easily  moved to heat.DNDarrays I think

Reply:
> You use a lot of torch.Tensors, which is not necessary as far as I see. Would be better to transfer this functionality to ht.DNDarrays as far as possible and call _DNDarray__array if the local torch tensor is required

Hmm, I can see that it might make things easier to follow when reading the code. But there are lots of instances where we would have to wrap a torch tensor into a DNDarray, just to use the underlying _DNDarray__array for the rest of the function. Why do that?

It would be good to settle this so we have a project-wide agreement. We could have a naming convention so one can tell torch and heat tensors apart at first glance.